c_reference/src/quantized_fastgrnn.c

```diff
@@ -204,12 +213,21 @@ int q15_fastgrnn(Q15_T* const hiddenState, ITER_T hiddenDims,
     }
 
     // Process the new input and previous hidden state
-    q15_m_mulvec(tparams->W, tbuffers->normFeatures, hiddenDims, inputDims,
-      tbuffers->preComp1, tscales->W, tscales->normFeaturesMVW, tscales->H1W,
-      tscales->H2W);
-    q15_m_mulvec(tparams->U, hiddenState, hiddenDims, hiddenDims,
-      tbuffers->preComp2, tscales->U, tscales->hiddenStateMVU, tscales->H1U,
-      tscales->H2U);
+    #ifdef SPARSE
```

Comment:
should we interleave sparse code like this or separate out to another set of files?

Reply:
As far as `RNNPool` is concerned, these two calls are the only two sparsity related changes needed. Separating out to another set of files would be wasteful in my opinion.